src/time/span.cr

```diff
@@ -263,33 +323,39 @@ struct Time::Span
     io << '0' if seconds < 10
     io << seconds
 
-    fractional = ticks.remainder(TicksPerSecond).abs.to_i32
-    if fractional != 0
+    nanoseconds = self.nanoseconds.abs
+    if nanoseconds != 0
       io << '.'
-      io << '0' if fractional < 1000000
-      io << '0' if fractional < 100000
-      io << '0' if fractional < 10000
-      io << '0' if fractional < 1000
-      io << '0' if fractional < 100
-      io << '0' if fractional < 10
-      io << fractional
+      io << '0' if nanoseconds < 100000000
+      io << '0' if nanoseconds < 10000000
+      io << '0' if nanoseconds < 1000000
+      io << '0' if nanoseconds < 100000
+      io << '0' if nanoseconds < 10000
+      io << '0' if nanoseconds < 1000
+      io << '0' if nanoseconds < 100
+      io << '0' if nanoseconds < 10
+      io << nanoseconds
     end
   end
 
-  def self.from(value, tick_multiplicator) : self
-    # TODO check nan
-    # TODO check infinity and overflow
-    value = value * (tick_multiplicator / TicksPerMillisecond)
-    val = (value < 0 ? (value - 0.5) : (value + 0.5)).to_i64 # round away from zero
-    Span.new(val * TicksPerMillisecond)
+  # :nodoc:
+  def self.from(value, nanos_multiplicator) : self
```

Comment:
This method may be the most problematic, it takes a maximum Int64::MAX nanoseconds *after* the multiplicator is applied, which is very far from being capable to render the full potential of Time::Span representations.

Most regular use cases won't overflow, but larger computations will, e.g. `106752.days`.

To be honest, the overflow was already present, just with slightly larger values, e.g. `10675200.days` but we raise an exception (Time::Span too big or to small). I have a feeling the overflow is now silenced?

Reply:
I may be totally wrong, here, since `Time::Span.from` is only ever called from `Float.seconds`, ... for example. `Int#days` is fine.

Reply:
Indeed. I'll move the method to Float and make it private, it makes more sense :-)